test/runtests.jl

```diff
@@ -380,22 +380,22 @@ end
 # type, and produce numerically correct results.
 as_array(x::AbstractArray) = Array(x)
 as_array(x::UniformScaling) = x
-equal_or_undef(a::Number, b::Number) = (a == b) || isequal(a, b)
-equal_or_undef(a, b) = all(equal_or_undef.(a, b))
+isapprox_or_undef(a::Number, b::Number) = (a ≈ b) || isequal(a, b)
```

Comment:
this is fine.
I had though might want to flip order because of types that mightt define `isapprox` but would define `isequal`.
But if you restrict to `Number` subtypes there is a fallback defintion for `isapprox` that does call `==` at least.
And I think that has you covered